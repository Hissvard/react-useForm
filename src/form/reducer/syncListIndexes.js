export const syncListIndexes = (state, fieldsKey, path) => {
  const items = state.getIn([fieldsKey, ...path, 'items'])
    .map((item, index) => {
      const fieldKeys = item.get('fields').keySeq()
      let temp = item
      for (let key of fieldKeys) {
        temp = temp.setIn(['fields', key, 'initial', 'field', 'name'], `${path.join('.')}.items.${index}.fields.${key}`)
      }
      return temp
    })

  return state.setIn([fieldsKey, ...path, 'items'], items)
}
