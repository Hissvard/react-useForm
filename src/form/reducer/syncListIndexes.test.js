import { fromJS } from 'immutable'
import { syncListIndexes } from './syncListIndexes'

let state, fieldsKey, path
beforeEach(() => {
  fieldsKey = 'fields'
  path = ['nested', 'data']
  state = fromJS({
    [fieldsKey]: {
      nested: {
        data: {
          items: [
            {
              fields: {
                first: {
                  initial: {
                    field: {
                      name: 'nested.data.items.1.fields.first',
                    },
                  },
                },
                second: {
                  items: [
                    {
                      fields: {
                        shade: {
                          initial: {
                            field: {
                              name: 'nested.data.items.1.fields.second.items.3.fields.shade',
                            },
                          },
                        },
                      },
                    }, {
                      fields: {
                        shade: {
                          initial: {
                            field: {
                              name: 'nested.data.items.1.fields.second.items.5.fields.shade',
                            },
                          },
                        },
                      },
                    },
                  ],
                  initial: {
                    field: {
                      name: 'nested.data.items.1.fields.second',
                    },
                  },
                },
              },
            },
            {
              fields: {
                first: {
                  initial: {
                    field: {
                      name: 'nested.data.items.4.fields.first',
                    },
                  },
                },
                second: {
                  initial: {
                    field: {
                      name: 'nested.data.items.4.fields.second',
                    },
                  },
                },
              },
            },
          ],
        },
      },

    },
  })
})


test('resets all index to 0 index', () => {
  expect(syncListIndexes(state, fieldsKey, path)).toMatchSnapshot()
})
