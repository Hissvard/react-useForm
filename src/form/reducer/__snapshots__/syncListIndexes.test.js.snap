// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resets all index to 0 index 1`] = `
Immutable.Map {
  "fields": Immutable.Map {
    "nested": Immutable.Map {
      "data": Immutable.Map {
        "items": Immutable.List [
          Immutable.Map {
            "fields": Immutable.Map {
              "first": Immutable.Map {
                "initial": Immutable.Map {
                  "field": Immutable.Map {
                    "name": "nested.data.items.0.fields.first",
                  },
                },
              },
              "second": Immutable.Map {
                "initial": Immutable.Map {
                  "field": Immutable.Map {
                    "name": "nested.data.items.0.fields.second",
                  },
                },
              },
            },
          },
          Immutable.Map {
            "fields": Immutable.Map {
              "first": Immutable.Map {
                "initial": Immutable.Map {
                  "field": Immutable.Map {
                    "name": "nested.data.items.1.fields.first",
                  },
                },
              },
              "second": Immutable.Map {
                "initial": Immutable.Map {
                  "field": Immutable.Map {
                    "name": "nested.data.items.1.fields.second",
                  },
                },
              },
            },
          },
        ],
      },
    },
  },
}
`;
